### DISCUSSION TITLE
Design for preparing SNOS input

### DISCUSSION THREAD

**Discussion by tdelabro** [keep-starknet-strange/madara#1582] [Ideas]
What should be our design while using the state changes to prepare a SNOS input to give to the entity responsible for taking on the next steps?

Should it even be a SNOS input, or something else?

**Comment by hhamud** [comment]
What does SNOS input mean?

**Reply by tdelabro** [reply]
It is the struct containing the data expected by the SNOS (Starknet OS). The SNOS run is the provable run of the block. It's the entrypoint of the proving pipeline

**Comment by tdelabro** [comment]
defined in #1595
